syntax = "proto3";

package user;

service User {
	rpc GetUsers (UserListRequest) returns (UserListResponse);

	rpc GetUser (UserParamRequest) returns (UserResponse);

	rpc CreateUser (UserRequest) returns (UserResponse);

	rpc UpdateUser (UserRequest) returns (UserResponse);

	rpc DeleteUser (UserParamRequest) returns (UserParamRequest);
}

message UserParamRequest {
  int32 id = 1;
}

message UserRequest {
  string first_name = 1;
  string last_name = 2;
  string phone = 3;
  string address = 4;
}

message UserResponse {
  int32 id = 1;
  string first_name = 2;
  string last_name = 3;
  string phone = 4;
  string address = 5;
  string created_date = 6;
  string created_by = 7;
  string updated_date = 8;
  string updated_by = 9;
}

message UserListRequest {
  string search = 1;
  int32 page = 2;
  int32 size = 3;
}

message UserListResponse {
  repeated UserResponse items = 1;
  int32 page = 2;
  int32 total = 3;
  int32 page_size = 4;
}